### DISCUSSION TITLE
backend server failing to start on docker

### DISCUSSION THREAD

**Discussion by koti-buchepalli** [hoppscotch/hoppscotch#2993] [Q&A]
Steps: Just cloned the repo and updated .env file and executed docker compose up hoppscotch-sh-admin

docker version : Docker version 20.10.23, build 7155243

Logs for backend on docker container :
=============================
hoppscotch-backend@2023.4.0 start:prod /usr/src/app
2023-04-16 07:44:35 > node dist/main
2023-04-16 07:44:35 
2023-04-16 07:44:35 node:internal/modules/cjs/loader:958
2023-04-16 07:44:35   throw err;
2023-04-16 07:44:35   ^
2023-04-16 07:44:35 
2023-04-16 07:44:35 Error: Cannot find module '/usr/src/app/dist/main'
2023-04-16 07:44:35     at Module._resolveFilename (node:internal/modules/cjs/loader:955:15)
2023-04-16 07:44:35     at Module._load (node:internal/modules/cjs/loader:803:27)
2023-04-16 07:44:35     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
2023-04-16 07:44:35     at node:internal/main/run_main_module:18:47 {
2023-04-16 07:44:35   code: 'MODULE_NOT_FOUND',
2023-04-16 07:44:35   requireStack: []
2023-04-16 07:44:35 }
2023-04-16 07:44:35 
2023-04-16 07:44:35 Node.js v18.8.0
2023-04-16 07:44:35  ELIFECYCLE  Command failed with exit code 1.

I tried to execute on my local machine as well but there were different errors so I thought I would use docker to keep it simple , Any thoughts ?

**Comment by koti-buchepalli** [comment]
Any thoughts on this @liyasthomas

**Reply by koti-buchepalli** [reply]
@balub As I was not sure if this is a bug so created a discussion. I am still facing same issue. I see you suggested pnpm i before docker compose up -d , Not really convinced but tried on my mac and it popped up with an another error , Thank you for your time :) 

 hoppscotch main ✗ ❯ pnpm i
Scope: all 10 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
 WARN  Local dependency not found at /Users/Play/hoppscotch/hoppscotch/packages/hoppscotch-backend/@types/nestjs/jwt
packages/codemirror-lang-graphql prepare$ rollup -c
│ Browserslist: caniuse-lite is outdated. Please run:
│   npx browserslist@latest --update-db
│   Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
│ 
│ src/index.js → dist/index.cjs, ./dist...
│ [!] (plugin Typescript) TypeError: path.charCodeAt is not a function
│ src/index.js
│ TypeError: path.charCodeAt is not a function
│     at getEncodedRootLength (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/typesc
│     at Object.isRootedDiskPath (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/typ
│     at Object.isExternalModuleNameRelative (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node
│     at tryLoadModuleUsingOptionalResolutionSettings (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4
│     at tryResolve (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/typescript/lib/t
│     at /Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/typescript/lib/typescript.js
│     at Object.forEach (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/typescript/l
│     at nodeModuleNameResolverWorker (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_module
│     at nodeModuleNameResolver (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/type
│     at Object.resolveModuleName (/Users/Play/hoppscotch/hoppscotch/node_modules/.pnpm/typescript@4.7.4/node_modules/ty
└─ Failed in 2.3s at /Users/Play/hoppscotch/hoppscotch/packages/codemirror-lang-graphql
packages/hoppscotch-backend postinstall$ prisma generate && pnpm run generate-gql-sdl
│ Prisma schema loaded from prisma/schema.prisma
└─ Running...
packages/hoppscotch-data prepare$ pnpm run build:code && pnpm run build:decl
│ > @hoppscotch/data@0.4.4 build:code /Users/Play/hoppscotch/hoppscotch/packages/hoppscotch-data
│ > vite build
│ vite v3.2.4 building for production...
│ transforming...
│ ✓ 287 modules transformed.
│ rendering chunks...
│ dist/hoppscotch-data.js   46.28 KiB / gzip: 13.40 KiB
│ dist/hoppscotch-data.cjs   34.28 KiB / gzip: 12.04 KiB
└─ Running...
packages/hoppscotch-ui postinstall$ pnpm run build
│ > @hoppscotch/ui@0.0.1 build /Users/Play/hoppscotch/hoppscotch/packages/hoppscotch-ui
│ > vite build
└─ Running...
. prepare$ husky install
│ husky - Git hooks installed
└─ Done in 468ms
 ELIFECYCLE  Command failed with exit code 1.

**Comment by imdiode** [comment]
I am facing the same issue

**Comment by TVW96** [comment]
Running `pnpm install` on the project is giving me the same issue. 

'MODULE_NOT_FOUND'

ELIFECYCLE Command failed with exit code 1.
